### PR TITLE
Fix flaky sidebar toggles

### LINKED ISSUE #5411: Upper button to toggle right sidebar disappears after closing the sidebar
## Context

In a newly opened annotation

## Expected Behavior

The upper button to toggle the right sidebar should remain visible and usable after closing the right sidebar

## Current Behavior

The button disappears, see gif

close-right-sidebar

## Steps to Reproduce the bug

- [ ] Cannot reproduce the bug anymore / needs deeper investigation.

1. Open a new annotation
2. Use the upper right sidebar toggle button to close the sidebar
3. The button is no longer visible
4. (Compare this to the upper left sidebar toggle which remains visible after closing)

## Your Environment for bug

* Browser name and version: Chrome 90
* Operating System and version: Ubuntu 18.04
* Version of WebKnossos (Release or Commit): : 21.02.0-14379

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Open a new annotation
- Use the upper right sidebar toggle button to close the sidebar
- Use "." to maximize/unmaximize viewports
- The sidebar toggle buttons should always stay visible

### Issues:
- fixes #5411
- while testing, I also noticed another not-ideal behavior for which I created https://github.com/scalableminds/webknossos/issues/5413

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/view/layouting/flex_layout_wrapper.js [hunk 1]
```diff
@@ -336,6 +335,7 @@ class FlexLayoutWrapper extends React.PureComponent<Props, State> {
       // Only adjust the internal state if the toggle is not automated and no tab is maximized.
       this.borderOpenStatusWhenNotMaximized[side] = !this.borderOpenStatusWhenNotMaximized[side];
     }
+    this.state.model.doAction(FlexLayout.Actions.selectTab(`${side}-border-tab-container`));
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/layouting/flex_layout_wrapper.js:338]: Performing this action triggers a re-render (on slower system this is probably delayed which is why the bug didn't occur). While re-rendering the border status from the store is used (which is updated in the lines above). Therefore, moving this action made sense to me.

**Comment by MichaelBuessemeyer** [comment]
I just pushed a commit that (hopefully) tackles this bug's root cause.
@philippotto and I found out, that the bug was occurring because in the `getPositionStatusOf`-method that is used to determine whether a tablet should have a border toggle button or not had incorrect information about which borders / sidebars are open at the moment.
=> The newest version of the `getPositionStatusOf`-method does not need the information which borders are open anymore. It simply compares the tabs extends / rect with the root container's extends / rect. Additionally, this made the whole code easier.

Note that the root container does not include the borders / sidebars

File: frontend/javascripts/oxalis/view/layouting/flex_layout_wrapper.js (hunk 1 above)
Comment (MichaelBuessemeyer) [reply frontend/javascripts/oxalis/view/layouting/flex_layout_wrapper.js:338]: If the newest fix (see below) fixes this bug, this change is not necessary anymore. But I think we can just keep it.

**Comment by philippotto** [comment]
@MichaelBuessemeyer awesome fix! feel free to accept the PR (since I opened it, I cannot do it) and merge it :rocket:

**Review by MichaelBuessemeyer — APPROVED** [review]

**Pull request merged**